src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1019,20 +1019,15 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 		const description = nls.localize('TerminalTaskSystem.terminalDescription', 'Task');
 		let originalCommand = task.command.name;
 		if (isShellCommand) {
-			let defaultConfig: { shell: string, args: string[] | string | undefined };
-			if (variableResolver.taskSystemInfo) {
-				defaultConfig = await variableResolver.taskSystemInfo.getDefaultShellAndArgs();
-			} else {
-				const defaultProfile = await this.terminalProfileResolverService.getDefaultProfile({
-					allowAutomationShell: true,
-					os: Platform.OS,
-					remoteAuthority: this.environmentService.remoteAuthority
-				});
-				defaultConfig = {
-					shell: defaultProfile.path,
-					args: defaultProfile.args
-				};
-			}
+			const defaultProfile = await this.terminalProfileResolverService.getDefaultProfile({
+				allowAutomationShell: true,
+				os: Platform.OS,
```

Comment:
@Tyriar `Platform.OS` could be the wrong OS if connected to a remote. Will `getDefaultProfile` ignore the wrong platform in that case?

Reply:
Yep, tasks when connected to a remote are broken.